extensions/auth/portability-auth-google/src/main/java/org/dataportabilityproject/auth/google/GoogleAuthDataGenerator.java

```diff
@@ -0,0 +1,138 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.dataportabilityproject.auth.google;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.api.client.auth.oauth2.AuthorizationCodeFlow;
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.auth.oauth2.TokenResponse;
+import com.google.api.client.googleapis.auth.oauth2.GoogleAuthorizationCodeFlow;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.common.base.Charsets;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableCollection;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.io.BaseEncoding;
+import java.io.IOException;
+import org.dataportabilityproject.spi.gateway.auth.AuthDataGenerator;
+import org.dataportabilityproject.spi.gateway.types.AuthFlowConfiguration;
+import org.dataportabilityproject.types.transfer.auth.AuthData;
+import org.dataportabilityproject.types.transfer.auth.TokensAndUrlAuthData;
+/**
+ * Provides configuration for conducting an OAuth flow against the Google AD API. Returned tokens
+ * can be used to make requests against the Google Graph API.
+ *
+ * <p>The flow is a two-step process. First, the user is sent to an authorization page and is then
+ * redirected to a address in this system with the authorization code. The second step takes the
+ * authorization code and posts it against the AD API to obtain a token for querying the Graph API.
+ */
+public class GoogleAuthDataGenerator implements AuthDataGenerator {
+  // TODO: Reduce requested scopes by service
+  private static final ImmutableCollection<String> SCOPES =
+      ImmutableSet.of("user.read", "mail.read", "Contacts.ReadWrite", "Calendars.ReadWrite");
+
+  private final String redirectPath;
+  private final String clientId;
+  private final String clientSecret;
+  private final HttpTransport httpTransport;
+  private final ObjectMapper objectMapper;
+
+  /**
+   * @param redirectPath the path part this generator is configured to request OAuth authentication
+   *     code responses be sent to
+   * @param clientId The Application ID that the registration portal (apps.dev.microsoft.com)
+   *     assigned the portability instance
+   * @param clientSecret The application secret that was created in the app registration portal for
+   *     the portability instance
+   * @param httpTransport The http transport to use for underlying GoogleAuthorizationCodeFlow
+   * @param objectMapper The json factory provider
+   */
+  public GoogleAuthDataGenerator(
+      String redirectPath,
+      String clientId,
```

Comment:
This should take an AppCredential

Reply:
just noticed that you pass in the id and secret from the AppCredential. It probably doesnt matter but can we change this or the MSAuthExtension to be consistent? 

https://github.com/google/data-transfer-project/blob/125cfb57b8380aad21b6bff0afe62855c4f70501/extensions/auth/portability-auth-microsoft/src/main/java/org/dataportabilityproject/auth/microsoft/MicrosoftAuthDataGenerator.java#L49

Reply:
Cool, let's refactor in a follow up PR if we want it.